src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,232 @@
+                }
+
+                AddProjectToProjectReference(project, frameworkOption.Value(), referenceOption.Values);
+
+                project.Save();
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message);
+                return 1;
+            }
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
+        // I.e. it returns null for some valid projects
+        public static ProjectRootElement TryOpenProject(string filename)
+        {
+            try
+            {
+                return ProjectRootElement.Open(filename);
+            }
+            catch (Microsoft.Build.Exceptions.InvalidProjectFileException)
+            {
+                return null;
+            }
+        }
+
+        public static ProjectRootElement GetProjectFromFileOrThrow(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                throw new GracefulException($"Provided project `{filename}` does not exist.");
+            }
+
+            var project = TryOpenProject(filename);
+            if (project == null)
+            {
+                throw new GracefulException($"Invalid MSBuild project `{filename}`.");
+            }
+
+            return project;
+        }
+
+        public static ProjectRootElement GetProjectFromCurrentDirectoryOrThrow()
+        {
+            DirectoryInfo currDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+            FileInfo[] files = currDir.GetFiles("*proj");
+            if (files.Length == 0)
+            {
+                throw new GracefulException("Could not find any MSBuild project in the current directory.");
+            }
+
+            if (files.Length > 1)
+            {
+                throw new GracefulException("Found more than one MSBuild project in the current directory. Please specify which one to use.");
+            }
+
+            FileInfo projectFile = files.First();
+
+            if (!projectFile.Exists)
+            {
+                throw new GracefulException("Could not find any project in the current directory.");
+            }
+
+            var ret = TryOpenProject(projectFile.FullName);
+            if (ret == null)
+            {
+                throw new GracefulException($"Found an MSBuild project `{projectFile.FullName}` in the current directory but it is invalid.");
+            }
+
+            return ret;
+        }
+
+        public static Func<T, bool> AndPred<T>(params Func<T, bool>[] preds)
+        {
+            return (el) => preds.All((pred) => pred == null || pred(el));
+        }
+
+        public static string GetFrameworkConditionString(string framework)
+        {
+            if (string.IsNullOrEmpty(framework))
+            {
+                return null;
```

Comment:
null/empty are the same in my eyes and both mean: I don't want a condition. I don't want a condition == null or "". Since I believe that no condition is as valid as the actual condition - it is easier to do the check in a single place and use the same predicate for "this means no condition" than do it all over the code. I.e. if we assume that GetFrameworkConditionString will throw on null/empty, now I got to know what predicate that function used for "no framework" - is it "null" or null or "<none>" or ""? I will change this to TryGetFrameworkConditionString - that should feel more intuitive for both of us